### PR TITLE
Task/753 Randomize validator list

### LINKED ISSUE #753: Randomise the order of validators
The intention is to not bias the selection of validators.
Future iterations may contain sorts, or warning that help users select "Good" validators but not for now (and possibly even then, the default order should be random). 

The randomisation could happen on each load, but something that randomised on first load and then kept the order in that browser would be less janky for user.

more colour on this here, but I think this covers it: https://vegaprotocol.slack.com/archives/C01P24880JY/p1636385051158200

### PR DESCRIPTION
**Pull request by mattrussell36**
- Removes sorting logic for validators list (previously sorted by staked amounts)
- Uses lodash/shuffle to randomize the array list

Closes  #753

### CODE REVIEW

File: src/routes/staking/staking.tsx [hunk 1]
```diff
@@ -213,15 +214,9 @@ export const StakingStepSelectNode = ({
...
       };
     });
-    const sortedByStake = nodesWithPercentages.sort((a, b) => {
-      if (a.stakedOnNode.isLessThan(b.stakedOnNode)) return 1;
-      if (a.stakedOnNode.isGreaterThan(b.stakedOnNode)) return -1;
-      if (a.id < b.id) return 1;
-      if (a.id > b.id) return -1;
-      return 0;
-    });
+    const randomized = shuffle(nodesWithPercentages);
```
Comment (dexturr) [inline src/routes/staking/staking.tsx:217]: This changes the order every single time we refresh the validators, which could lead to people clicking on the wrong node.

**Review by dexturr — CHANGES_REQUESTED** [review]

File: src/routes/staking/staking.tsx (hunk 1 above)
Comment (mattrussell36) [reply src/routes/staking/staking.tsx]: @dexturr I've pushed quite a bit of new code here so worth having another look.

File: src/routes/staking/staking-nodes-container.tsx [hunk 2]
```diff
@@ -71,24 +71,52 @@ export const StakingNodesContainer = ({
...
+      const now = Date.now();
+      const expiry = new Date(data.epoch.timestamps.expiry).getTime();
+
+      console.table({
+        now: new Date(now).toISOString(),
+        expiry: new Date(expiry).toISOString(),
+      });
+
+      if (now > expiry) {
+        console.log("refetching");
```
Comment (dexturr) [inline src/routes/staking/staking-nodes-container.tsx:94]:
```suggestion
```

File: src/routes/staking/staking-nodes-container.tsx [hunk 3]
```diff
@@ -71,24 +71,52 @@ export const StakingNodesContainer = ({
...
+  React.useEffect(() => {
+    const interval = setInterval(() => {
+      if (!data?.epoch.timestamps.expiry) return;
+      const now = Date.now();
+      const expiry = new Date(data.epoch.timestamps.expiry).getTime();
+
+      console.table({
+        now: new Date(now).toISOString(),
+        expiry: new Date(expiry).toISOString(),
+      });
```
Comment (dexturr) [inline src/routes/staking/staking-nodes-container.tsx:91]:
```suggestion
```

**Review by dexturr — APPROVED** [review]
I like this solution a lot :D

**Pull request merged**